numpy/lib/function_base.py

```diff
@@ -4839,6 +4843,26 @@ def digitize(x, bins, right=False):
     if mono == 0:
         raise ValueError("bins must be monotonically increasing or decreasing")
 
+    if edge:
+        # =========  =============  ============================ ===== =====
+        # `right`    order of bins  returned index `i` satisfies delta index
+        # =========  =============  ============================ ===== =====
+        # ``False``  increasing     ``bins[i-1] <= x < bins[i]``   1    -1
+        # ``True``   increasing     ``bins[i-1] < x <= bins[i]``   -1    0
+        # ``False``  decreasing     ``bins[i-1] > x >= bins[i]``   1    0
+        # ``True``   decreasing     ``bins[i-1] >= x > bins[i]``   -1    -1
+        # =========  =============  ============================ ===== =====
+        delta = -1 if right else 1
+        idx = -1 if delta == mono else 0
+        if np.issubdtype(bins.dtype, _nx.integer):
+            bins = _nx.int64(bins)
+            if abs(bins[idx]) >= 2 ** 63 - 1:
```

Comment:
This rejects uint64 `2**64-2`, but should not.

Reply:
`_nx.int64(2 ** 63)` throws an error but `_nx.int64(2 ** 63 - 1)` does not so as far as I know that's the top of the range. That seems like a pretty big range of acceptable numbers, I think that should do it if that's the range of `int64`, not sure it would be worth extending to larger numbers unless there is some particular reason.

Reply:
My comment is about `uint64`

Reply:
I'm running into some real funny behavior with `bins = np.uint64([0, 2**64 - 2])` where `bins[-1] += 1` gives `[0, 0]` as does `bins[-1] -= 1`. And so does `bins[-1] = np.uint64(bins[-1] + 1)`. Very strange.

This should definitely not be figured out in np.digitize and should be it's own issue. For digitize, I'd be in favor of just leaving it as is and not supporting the very narrow use-case of unsigned integers above 2 ** 63 at least until that gets figured out elsewhere.

Reply:
Right, the issue you're seeing is that `1` is of type `int64`. `bins[i] += np.uint64(1)` should work for the uint64 case, although will fail for the `int64` case.

Reply:
Thanks for the help, okay now this should work for `uint64`